internal/discovery/services_discoverer.go

```diff
@@ -0,0 +1,53 @@
+package discovery
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ServicesLister interface {
+	List(selector labels.Selector) (ret []*corev1.Service, err error)
+}
+
+type ServiceDiscoverer interface {
+	Discover() ([]*corev1.Service, error)
+}
+
+type servicesDiscoverer struct {
+	ServicesLister ServicesLister
+}
+
+func (d *servicesDiscoverer) Discover() ([]*corev1.Service, error) {
+
+	services, err := d.ServicesLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("listing services: %w", err)
+	}
+
+	return services, nil
+}
+
+func NewServicesDiscoverer(client kubernetes.Interface) ServiceDiscoverer {
```

Comment:
Since we never pass any config whatsoever to `ServicesDiscoverer`, I don't see how it can "discover" anything. Instead, it just _lists_. Shouldn't we then use the `ServiceLister` directly?